### ISSUE TITLE
Socket in library really slow for GetChannelList request? (over 100 channels)

### ISSUE THREAD

**Issue by qvazzler** [Scordo/TS3QueryLib.Net#12]
Hi.

I'm experiencing issues with the socket used in this library when I need to obtain a large set of information.

After doing the command manually in serverquery through putty and saving the result to a txt file; I can acknowledge that the information amounts to roughly 25kb size.

I've tried to alter the RECEIVE_BUFFER_SIZE, but no luck.

Anyone have any advice on this? This is the data I'm supposed to receive

```
cid=10994 pid=0 channel_order=0 channel_name=[*spacer616]• total_clients=0 channel_needed_subscribe_power=0|cid=414 pid=0 channel_order=10994 channel_name=Noble\sRoom total_clients=0 channel_needed_subscribe_power=0|cid=29845 pid=0 channel_order=414 channel_name=[*spacer110]\sˑ total_clients=0 channel_needed_subscribe_power=0|cid=22460 pid=0 channel_order=29845 channel_name=Noblesse\sRoom total_clients=0 channel_needed_subscribe_power=0|cid=274 pid=0 channel_order=22460 channel_name=[*spacer66]• total_clients=0 channel_needed_subscribe_power=0|cid=413 pid=0 channel_order=274 channel_name=[spacer88]↓\sPublic\sSpace\s↓ total_clients=0 channel_needed_subscribe_power=0|cid=10988 pid=0 channel_order=413 channel_name=[*spacer999994]ː total_clients=0 channel_needed_subscribe_power=0|cid=516 pid=0 channel_order=10988 channel_name=Open\sRooms total_clients=2 channel_needed_subscribe_power=0|cid=42387 pid=516 channel_order=0 channel_name=Room\s1 total_clients=1 channel_needed_subscribe_power=0|cid=124 pid=0 channel_order=516 channel_name=[*spacer10]\sˑ total_clients=0 channel_needed_subscribe_power=0|cid=10989 pid=0 channel_order=124 channel_name=Private\sRooms total_clients=0 channel_needed_subscribe_power=0|cid=29372 pid=10989 channel_order=0 channel_name=iShrek total_clients=0 channel_needed_subscribe_power=0|cid=37767 pid=10989 channel_order=29372 channel_name=Sthlm total_clients=0 channel_needed_subscribe_power=0|cid=41097 pid=10989 channel_order=37767 channel_name=PsychO's\soratorical\sroom total_clients=0 channel_needed_subscribe_power=0|cid=7423 pid=0 channel_order=10989 channel_name=[*spacer123]• total_clients=0 channel_needed_subscribe_power=0|cid=10983 pid=0 channel_order=7423 channel_name=[spacer99999991]↓\sPopular\sGames\s↓ total_clients=0 channel_needed_subscribe_power=0|cid=10985 pid=0 channel_order=10983 channel_name=[*spacer999993]ː total_clients=0 channel_needed_subscribe_power=0|cid=1405 pid=0 channel_order=10985 channel_name=Dota\s2 total_clients=0 channel_needed_subscribe_power=0|cid=218 pid=1405 channel_order=0 channel_name=Looking\sfor\sGroup total_clients=0 channel_needed_subscribe_power=0|cid=11389 pid=218 channel_order=0 channel_name=[\sJust\sfor\sFun\s] total_clients=0 channel_needed_subscribe_power=0|cid=11387 pid=218 channel_order=11389 channel_name=[\sTrash\s] total_clients=0 channel_needed_subscribe_power=0|cid=24926 pid=218 channel_order=11387 channel_name=[\sLowest\s] total_clients=0 channel_needed_subscribe_power=0|cid=7168 pid=218 channel_order=24926 channel_name=[\sLower\s] total_clients=0 channel_needed_subscribe_power=0|cid=222 pid=218 channel_order=7168 channel_name=[\sLow\s] total_clients=0 channel_needed_subscribe_power=0|cid=223 pid=218 channel_order=222 channel_name=[\sNormal\s] total_clients=0 channel_needed_subscribe_power=0|cid=24927 pid=218 channel_order=223 channel_name=[\sNormal\s#2\s] total_clients=0 channel_needed_subscribe_power=0|cid=24928 pid=218 channel_order=24927 channel_name=[\sNormal\s#3\s] total_clients=0 channel_needed_subscribe_power=0|cid=224 pid=218 channel_order=24928 channel_name=[\sHigh\s] total_clients=0 channel_needed_subscribe_power=0|cid=7167 pid=218 channel_order=224 channel_name=[\sHigher\s] total_clients=0 channel_needed_subscribe_power=0|cid=11388 pid=218 channel_order=7167 channel_name=[\sHighest\s] total_clients=0 channel_needed_subscribe_power=0|cid=11390 pid=218 channel_order=11388 channel_name=[\sTeam\sMatchmaking\s] total_clients=0 channel_needed_subscribe_power=0|cid=225 pid=1405 channel_order=218 channel_name=Dota\s2\sClans total_clients=0 channel_needed_subscribe_power=0|cid=41040 pid=225 channel_order=0 channel_name=Long\sHair\sDon't\sCare total_clients=0 channel_needed_subscribe_power=0|cid=28870 pid=225 channel_order=41040 channel_name=Scelesti\sExitium total_clients=0 channel_needed_subscribe_power=0|cid=31106 pid=225 channel_order=28870 channel_name=Legendary\sEagles total_clients=0 channel_needed_subscribe_power=0|cid=32136 pid=225 channel_order=31106 channel_name=WhOaMi's\sbased\sgaming total_clients=0 channel_needed_subscribe_power=0|cid=33886 pid=225 channel_order=32136 channel_name=Wezoa total_clients=0 channel_needed_subscribe_power=0|cid=33745 pid=225 channel_order=33886 channel_name=егор\sлох total_clients=0 channel_needed_subscribe_power=0|cid=33081 pid=225 channel_order=33745 channel_name=0D\steam total_clients=0 channel_needed_subscribe_power=0|cid=7066 pid=225 channel_order=33081 channel_name=S-W'\s\s[[\sSCL\sTournament\s]] total_clients=0 channel_needed_subscribe_power=0|cid=28773 pid=225 channel_order=7066 channel_name=Desolation\sof\sEntropy total_clients=0 channel_needed_subscribe_power=0|cid=24252 pid=225 channel_order=28773 channel_name=Minepix total_clients=0 channel_needed_subscribe_power=0|cid=2904 pid=225 channel_order=24252 channel_name=dota\s2\syo total_clients=0 channel_needed_subscribe_power=0|cid=23134 pid=2904 channel_order=0 channel_name=dota\s2\syo\ssub-channel total_clients=0 channel_needed_subscribe_power=0|cid=34810 pid=225 channel_order=2904 channel_name=Weed\sGamers total_clients=0 channel_needed_subscribe_power=0|cid=30484 pid=225 channel_order=34810 channel_name=NewBorn^ total_clients=0 channel_needed_subscribe_power=0|cid=30934 pid=225 channel_order=30484 channel_name=pNb total_clients=0 channel_needed_subscribe_power=0|cid=32714 pid=225 channel_order=30934 channel_name=Heed\sthe\sCall total_clients=0 channel_needed_subscribe_power=0|cid=33401 pid=225 channel_order=32714 channel_name=Ronnin\sGeek's total_clients=0 channel_needed_subscribe_power=0|cid=33440 pid=225 channel_order=33401 channel_name=FFB total_clients=0 channel_needed_subscribe_power=0|cid=33455 pid=225 channel_order=33440 channel_name=KOT total_clients=0 channel_needed_subscribe_power=0|cid=33531 pid=225 channel_order=33455 channel_name=DoTA\sSwag total_clients=0 channel_needed_subscribe_power=0|cid=33750 pid=225 channel_order=33531 channel_name=FAP total_clients=0 channel_needed_subscribe_power=0|cid=34037 pid=225 channel_order=33750 channel_name=Stijene total_clients=0 channel_needed_subscribe_power=0|cid=38160 pid=225 channel_order=34037 channel_name=Lethal\sLeague\sRoom total_clients=0 channel_needed_subscribe_power=0|cid=39103 pid=225 channel_order=38160 channel_name=best\sdoto\schannel total_clients=0 channel_needed_subscribe_power=0|cid=38659 pid=225 channel_order=39103 channel_name=Dota\s2\sby\svadik total_clients=0 channel_needed_subscribe_power=0|cid=39279 pid=225 channel_order=38659 channel_name=Melegség total_clients=0 channel_needed_subscribe_power=0|cid=40427 pid=225 channel_order=39279 channel_name=NITID total_clients=0 channel_needed_subscribe_power=0|cid=40584 pid=225 channel_order=40427 channel_name=Kappa\sGamers total_clients=0 channel_needed_subscribe_power=0|cid=41014 pid=225 channel_order=40584 channel_name=Anonimous total_clients=0 channel_needed_subscribe_power=0|cid=143 pid=1405 channel_order=225 channel_name=Room\s1 total_clients=0 channel_needed_subscribe_power=0|cid=34660 pid=1405 channel_order=143 channel_name=Dota\s2\s(provisional) total_clients=0 channel_needed_subscribe_power=0|cid=698 pid=0 channel_order=1405 channel_name=[*spacer19]• total_clients=0 channel_needed_subscribe_power=0|cid=10970 pid=0 channel_order=698 channel_name=[spacer9999999]↓\sSmall\sClans\s↓ total_clients=0 channel_needed_subscribe_power=0|cid=10971 pid=0 channel_order=10970 channel_name=[*spacer99991]ː total_clients=0 channel_needed_subscribe_power=0|cid=11195 pid=0 channel_order=10971 channel_name=General\sGaming\sClans total_clients=0 channel_needed_subscribe_power=0|cid=15258 pid=11195 channel_order=0 channel_name=AJSA\sOutlaws total_clients=0 channel_needed_subscribe_power=0|cid=32611 pid=11195 channel_order=15258 channel_name=GartenGamer total_clients=0 channel_needed_subscribe_power=0|cid=41965 pid=32611 channel_order=0 channel_name=World\sof\sTanks total_clients=0 channel_needed_subscribe_power=0|cid=41966 pid=32611 channel_order=41965 channel_name=Survarium total_clients=0 channel_needed_subscribe_power=0|cid=41967 pid=32611 channel_order=41966 channel_name=Dota2 total_clients=0 channel_needed_subscribe_power=0|cid=41968 pid=32611 channel_order=41967 channel_name=Hearthstone total_clients=0 channel_needed_subscribe_power=0|cid=29591 pid=11195 channel_order=32611 channel_name=Comrades\sOn\sDuty total_clients=0 channel_needed_subscribe_power=0|cid=20551 pid=11195 channel_order=29591 channel_name=CVT\sMembers\s\/\ssa-mp total_clients=0 channel_needed_subscribe_power=0|cid=11196 pid=11195 channel_order=20551 channel_name=UnknoowN total_clients=0 channel_needed_subscribe_power=0|cid=11200 pid=11196 channel_order=0 channel_name=AFK total_clients=0 channel_needed_subscribe_power=0|cid=22224 pid=11195 channel_order=11196 channel_name=SKUMRASKET total_clients=0 channel_needed_subscribe_power=0|cid=24588 pid=11195 channel_order=22224 channel_name=Annihilated total_clients=0 channel_needed_subscribe_power=0|cid=24640 pid=24588 channel_order=0 channel_name=WoW total_clients=0 channel_needed_subscribe_power=0|cid=37760 pid=24588 channel_order=24640 channel_name=Room\s2 total_clients=0 channel_needed_subscribe_power=0|cid=25554 pid=11195 channel_order=24588 channel_name=FUGLY\sDOT\sCOM total_clients=0 channel_needed_subscribe_power=0|cid=26962 pid=11195 channel_order=25554 channel_name=That\sFeel total_clients=0 channel_needed_subscribe_power=0|cid=27374 pid=11195 channel_order=26962 channel_name=Chikadori\sRoom total_clients=0 channel_needed_subscribe_power=0|cid=28487 pid=11195 channel_order=27374 channel_name=Malcanski\sSektasi total_clients=0 channel_needed_subscribe_power=0|cid=32481 pid=11195 channel_order=28487 channel_name=NiceguyZ total_clients=0 channel_needed_subscribe_power=0|cid=35615 pid=11195 channel_order=32481 channel_name=Záhradníctvo total_clients=0 channel_needed_subscribe_power=0|cid=41651 pid=11195 channel_order=35615 channel_name=Mycket\scancer\svar\sdet\shär total_clients=0 channel_needed_subscribe_power=0|cid=30582 pid=0 channel_order=11195 channel_name=Badass\sBreakfast\sClub total_clients=0 channel_needed_subscribe_power=0|cid=11198 pid=0 channel_order=30582 channel_name=[*spacer999944]\sˑ total_clients=0 channel_needed_subscribe_power=0|cid=150 pid=0 channel_order=11198 channel_name=League\sof\sLegends total_clients=0 channel_needed_subscribe_power=0|cid=10976 pid=0 channel_order=150 channel_name=[*spacer99995]\sˑ total_clients=0 channel_needed_subscribe_power=0|cid=151 pid=0 channel_order=10976 channel_name=Counter-Strike:\sGO total_clients=0 channel_needed_subscribe_power=0|cid=41634 pid=151 channel_order=0 channel_name=gitarrkluben total_clients=0 channel_needed_subscribe_power=0|cid=22344 pid=151 channel_order=41634 channel_name=MMR\sCSGO\s(Perú) total_clients=0 channel_needed_subscribe_power=0|cid=23793 pid=151 channel_order=22344 channel_name=Info.Z.Dupy total_clients=0 channel_needed_subscribe_power=0|cid=30398 pid=23793 channel_order=0 channel_name=Won total_clients=0 channel_needed_subscribe_power=0|cid=31039 pid=151 channel_order=23793 channel_name=mephidota total_clients=0 channel_needed_subscribe_power=0|cid=28901 pid=151 channel_order=31039 channel_name=nickel\schannel total_clients=0 channel_needed_subscribe_power=0|cid=958 pid=151 channel_order=28901 channel_name=Tem\sVanita total_clients=0 channel_needed_subscribe_power=0|cid=17461 pid=151 channel_order=958 channel_name=[SALATA] total_clients=0 channel_needed_subscribe_power=0|cid=28713 pid=151 channel_order=17461 channel_name=iiuwr total_clients=0 channel_needed_subscribe_power=0|cid=30309 pid=151 channel_order=28713 channel_name=;) total_clients=0 channel_needed_subscribe_power=0|cid=31316 pid=151 channel_order=30309 channel_name=The\sAngry\sBomb total_clients=0 channel_needed_subscribe_power=0|cid=33391 pid=151 channel_order=31316 channel_name=NIWEERF total_clients=0 channel_needed_subscribe_power=0|cid=33496 pid=151 channel_order=33391 channel_name=Nilssons\shobbyrum total_clients=0 channel_needed_subscribe_power=0|cid=37566 pid=151 channel_order=33496 channel_name=S.I.R\sLounge total_clients=0 channel_needed_subscribe_power=0|cid=10978 pid=0 channel_order=151 channel_name=[*spacer99996]\sˑ total_clients=0 channel_needed_subscribe_power=0|cid=152 pid=0 channel_order=10978 channel_name=Counter-Strike:\sGlobal\sOffensive total_clients=0 channel_needed_subscribe_power=0|cid=8245 pid=0 channel_order=152 channel_name=[*spacer999]• total_clients=0 channel_needed_subscribe_power=0|cid=10965 pid=0 channel_order=8245 channel_name=[spacer99999]↓\sLarge\/Special\sClans\s↓ total_clients=0 channel_needed_subscribe_power=0|cid=10964 pid=0 channel_order=10965 channel_name=[*spacer9999]ː total_clients=0 channel_needed_subscribe_power=0|cid=242 pid=0 channel_order=10964 channel_name=Mumindalen total_clients=0 channel_needed_subscribe_power=0|cid=29 pid=0 channel_order=242 channel_name=[*spacer3]• total_clients=0 channel_needed_subscribe_power=0|cid=10967 pid=0 channel_order=29 channel_name=[spacer999999]↓\sAFK\sArea\s↓ total_clients=0 channel_needed_subscribe_power=0|cid=10968 pid=0 channel_order=10967 channel_name=[*spacer999999]ː total_clients=0 channel_needed_subscribe_power=0|cid=33 pid=0 channel_order=10968 channel_name=Away total_clients=0 channel_needed_subscribe_power=0|cid=32 pid=0 channel_order=33 channel_name=Busy total_clients=0 channel_needed_subscribe_power=0|cid=21745 pid=0 channel_order=32 channel_name=Forever\sAlone total_clients=0 channel_needed_subscribe_power=0|cid=10992 pid=0 channel_order=21745 channel_name=[spacer919]↓\sHow\sdo\sI\steamspeak?\s↓ total_clients=0 channel_needed_subscribe_power=0
error id=0 msg=ok
```

**Comment by Scordo** [comment]
Hi, 

can you please validate this for the new release?:
https://ts3querylib.codeplex.com/releases/view/628874

Thanks!
Scordo

**Comment by qvazzler** [comment]
@Scordo Unfortunately I will not be working with the framework until January at earliest. Also I kind of poked lots of holes in your existing code in my fork.. :-)

**Comment by Myszax** [comment]
Seems GetClientList got the same problem. With ~400 clients on server this command executing time is about 3-4 seconds or more sometimes. Compares to PHP it is slow, in TS3 PHP FRAMEWORK this time is not event half a second.

**Comment by Scordo** [comment]
@Myszax Is there any Chance to get a query login on the server to debug that? I dont have a server with so many users online :-(. So I can't identify the cause. Would be very cool to be able to check that.

**Comment by Myszax** [comment]
Please send me your IP to myszax14@gmail.com
I am using iptables for query port, so I have to add you.
It's sad because there is no PM on GitHub.

**Comment by Scordo** [comment]
If you clone the repositor, you'll find the mail in the log :-D.

I've sent you a mail.

**Comment by Scordo** [comment]
Please use AwaitableQueryDispatcher. It's much faster and fixes the issue. It's included in release 2.0
